File: java_solutions/412.fizz-buzz.java
/*
 * @lc app=leetcode id=412 lang=java
 *
 * [412] Fizz Buzz
 */

class Solution {
    public List<String> fizzBuzz(int n) {
        return IntStream.rangeClosed(1, n).mapToObj(i -> i % 15 == 0 ? "FizzBuzz" : i % 3 == 0 ? "Fizz" : i % 5 == 0 ? "Buzz" : String.valueOf(i)).collect(Collectors.toList());
    }
}

File: kotlin_solutions/412.fizz-buzz.kt
/*
 * @lc app=leetcode id=412 lang=kotlin
 *
 * [412] Fizz Buzz
 */

class Solution {fun fizzBuzz(n: Int): List<String> = (1..n).map { when { it % 15 == 0 -> {"FizzBuzz"} it % 3 == 0 -> {"Fizz"} it % 5 == 0 -> {"Buzz"} else -> it.toString() } }}
